python/python_fastapi_server/routers/edr_utils.py

```diff
@@ -0,0 +1,808 @@
+    )
+    locations_link = EDRQueryLink(
+        href=f"{base_url}/locations",
+        rel="data",
+        hreflang="en",
+        title="Locations query",
+        variables=locations_variables,
+    )
+
+    instances_link = None
+    if ref_times and len(ref_times) > 0:
+        instances_variables = Variables(
+            query_type="instances",
+            #  crs_details=[crs_object],
+            default_output_format="CoverageJSON",
+            output_formats=["CoverageJSON", "GeoJSON"],
+        )
+        instances_link = EDRQueryLink(
+            href=f"{base_url}/instances",
+            rel="collection",
+            hreflang="en",
+            title="Instances query",
+            variables=instances_variables,
+        )
+        data_queries = DataQueries(
+            position=EDRQuery(link=position_link),
+            cube=EDRQuery(link=cube_link),
+            instances=EDRQuery(link=instances_link),
+            locations=EDRQuery(link=locations_link),
+        )
+    else:
+        data_queries = DataQueries(
+            position=EDRQuery(link=position_link),
+            cube=EDRQuery(link=cube_link),
+            locations=EDRQuery(link=locations_link),
+        )
+
+    parameter_names = get_params_for_collection(
+        edr_collection=edr_collection, wmslayers=wmslayers
+    )
+
+    crs = ["EPSG:4326"]
+
+    output_formats = ["CoverageJSON"]
+    if instance is None:
+        collection = Collection(
+            links=links,
+            id=edr_collection,
+            extent=extent,
+            data_queries=data_queries,
+            parameter_names=parameter_names,
+            crs=crs,
+            output_formats=output_formats,
+        )
+    else:
+        collection = Instance(
+            links=links,
+            id=instance,
+            extent=extent,
+            data_queries=data_queries,
+            parameter_names=parameter_names,
+            crs=crs,
+            output_formats=output_formats,
+        )
+
+    return collection, ttl
+
+
+def parse_period_string(period: str):
+    if period.startswith("PT"):
+        pattern = re.compile(r"""PT(\d+)([HMS])""")
+        match = pattern.match(period)
+        step = int(match.group(1))
+        step_type = match.group(2).lower()
+    else:
+        pattern = re.compile(r"""P(\d+)([YMD])""")
+        match = pattern.match(period)
+        step = int(match.group(1))
+        step_type = match.group(2)
+
+    if step_type == "Y":
+        delta = relativedelta(years=step)
+    elif step_type == "M":
+        delta = relativedelta(months=step)
+    elif step_type == "D":
+        delta = relativedelta(days=step)
+    elif step_type == "h":
+        delta = relativedelta(hours=step)
+    elif step_type == "m":
+        delta = relativedelta(minutes=step)
+    elif step_type == "s":
+        delta = relativedelta(seconds=step)
+    return delta
+
+
+def parse_interval_string(time_interval: str, ref_time: datetime):
```

Comment:
Can you provide a small descriptive docstring for this function?

Reply:
Done